src/Runner.Worker/StepsRunner.cs

```diff
@@ -352,7 +364,70 @@ private void CompleteStep(IStep step, TaskResult? result = null, string resultCo
         {
             var executionContext = step.ExecutionContext;
 
+            if (ShouldUploadAttachment(step))
+            {
+                QueueStepSummaryUpload(step);
```

Comment:
Should we error or warn the user if the step summary data size > our technical limitation of 128kb. Would be weird to have that data truncated with no indication. We could put it in the github.com docs, but more front and center to user has historically helped limit user friction.

Reply:
That can be done via the warning and error commands on the execution context of a step, like
`step.ExecutionContext.Error("The user will see this as an annotation")`

Reply:
Note that the above does not go into our trace/diag telemetry, so sometimes its nice to both Trace.Error and create an annotation

Reply:
+1 make sure you have the proper size limit before merging the PR.

Reply:
Do we need `try/catch` here?  This should be a best effort operation.  If failed, we should log an annotation but don't fail the job/step?